unfolder_test.go

```diff
@@ -574,9 +582,10 @@ func TestAllOk(t *testing.T) {
 		}))
 
 	mcr.AssertCalled(t, "ResolveContents",
-		mock.MatchedBy(func(actualDiffUuids map[string]bool) bool {
-			diffUuids[leadArticleUuid] = false
-			assert.Equal(t, diffUuids, actualDiffUuids)
+		mock.MatchedBy(func(actualDiffUuids []string) bool {
```

Comment:
This `mock.MatchedBy` can be refactored out as you copy/paste it several times here.

Reply:
I'm also not convinced by the logic here, why don't you just do `mcr.On("ResolveContents", expected).Return(...)` rather than allow any values and then assert the expectations at the end of the test?

Reply:
The line `mcr.On("ResolveContents", expected).Return(...)` would only declare the mock. If the method is not called the test will not fail. You would have no way of knowing if the method was actually called.

Reply:
It declares the mock, and then you conclude the test with `mcr.AssertExpectations` which verifies that all the declared mocks are called.